daemon/swap_daemon.go

```diff
@@ -158,8 +157,14 @@ func RunSwapDaemon(ctx context.Context, conf *SwapdConfig) (err error) {
 	})
 
 	log.Infof("starting swapd with data-dir %s", conf.EnvConf.DataDir)
-	err = rpcServer.Start() // blocks until server is shutdown or context is cancelled
-	if err != nil && !errors.Is(err, http.ErrServerClosed) {
+
+	err = rpcServer.Start()
+
+	if errors.Is(err, http.ErrServerClosed) {
+		// Returning nil to exit the program with exit code 0
+		// and to avoid a FATAL ERROR entry in log files
+		return nil
```

Comment:
Yeah, this needed a fix.  Instead of returning `nil` here, assign `err` to `nil`. The error handling in this function is a little delicate, as errors can be appended by defer statements above to track problems shutting down other subsystems.  We can't return `nil`, because then any multierror appended in the defer statements won't be seen by the caller.

Given that you were confused and might not be the last person to be confused, perhaps we should convert `return` statements in this function to be empty (or at least the final return statement)? This function was originally written without naming the return value and then converted, but I didn't change all the return statements during conversion. Use your best discretion about what will be most obvious to readers.